crates/stackable-versioned-macros/src/codegen/container/struct/k8s.rs

```diff
@@ -280,19 +280,19 @@ impl Struct {
             impl #enum_ident {
                 pub fn as_version_str(&self) -> &str {
                     match self {
-                        #(#variant_idents => #variant_strings),*
+                        #(#enum_ident::#variant_idents => #variant_strings),*
                     }
                 }
 
                 pub fn as_api_version_str(&self) -> &str {
                     match self {
-                        #(#variant_idents => #api_versions),*
+                        #(#enum_ident::#variant_idents => #api_versions),*
                     }
                 }
 
                 pub fn from_api_version(api_version: &str) -> Result<Self, #unknown_desired_api_version_error> {
                     match api_version {
-                        #(#api_versions => Ok(Self::#variant_idents)),*,
+                        #(#api_versions => Ok(#enum_ident::#variant_idents)),*,
```

Comment:
Any particular reason you prefer `#enum_ident` over `Self`?

Reply:
I think the concrete type should be used in generated code anyway.
But in person written code, I often use Self (mainly so that renames are easier to deal with).